sdk/storage/Azure.Storage.Common/src/Shared/ClientsideEncryption/IAuthenticatedCryptographicTransform.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Storage.Cryptography.Models;
+
+namespace Azure.Storage.Cryptography
+{
+    internal interface IAuthenticatedCryptographicTransform : IDisposable
```

Comment:
Could we get away with making these `abstract class`es instead?  It probably doesn't matter as much for shared source, but versioning is always more flexible with an `abstract class`.

Reply:
I'm actually a bit afraid of flexibility here for this shared source. This is delicate code and I wouldn't want any assumptions to be made when adding a new piece of functionality.